amazon-msk/cdk/secure-private-access/README.md

```diff
@@ -0,0 +1,390 @@
+
+#### List All CloudWatch Custom Metric Namespaces
+
+```bash
+aws cloudwatch list-metrics --query 'Metrics[*].Namespace' --output text | tr '\t' '\n' | sort | uniq | grep -v '^AWS'
+```
+
+In your `cdk.json` file add the desired CloudWatch Metrics Namespace for variable name `metrics.namespace` under `zilla-plus` object in the `cloudwatch` variables section.
+
+### Enable SSH Access
+
+To enable SSH access to the instances you will need the name of an existing EC2 KeyPair to set the `sshKey` context variable under `zilla-plus` object.
+
+List all EC2 KeyPairs:
+
+```bash
+aws ec2 describe-key-pairs --query 'KeyPairs[*].[KeyName]' --output table
+```
+
+Note down the KeyPair name `KeyName` you want to use.
+
+## Deploy stack using CDK
+
+### Install Project Dependencies
+
+Install the node.js dependencies specified in the `package.json` file:
+
+```bash
+npm install
+```
+
+### Synthesizing the CloudFormation Template
+
+Run the following command to synthesize your stack into a CloudFormation template:
+
+```bash
+cdk synth
+```
+
+This generates the cdk.out directory containing the synthesized CloudFormation template.
+
+### Bootstrap the environment (if needed)
+
+If this is your first time deploying in a specific AWS environment, bootstrap it:
+
+```bash
+cdk bootstrap
+```
+
+### Deploy the stack
+Deploy your resources to AWS:
+
+
+```bash
+cdk deploy
+```
+
+Sample outputs:
+```bash
+Outputs:
+SecurePrivateAccessStack.VpcEndpointServiceId = vpce-svc-1234567
+SecurePrivateAccessStack.VpcEndpointServiceName = com.amazonaws.vpce.us-east-1.vpce-svc-1234567
+Stack ARN:
+arn:aws:cloudformation:us-east-1:<account_id>:stack/SecurePrivateAccessStack/abcd1234
+```
+
+Once your stack is deployed, note down the VPC Endpoint Service Id and the VPC Endpoint Service Name, as you'll need this in the following steps when you add the VPC Endpoint from the client VPC.
+
+## Connect to your MSK Serverless from different VPC
+### Add a VPC endpoint in the client VPC
+You need to add a VPC Endpoint pointing to the Zilla Plus's VPC Endpoint Service in your client VPC. 
+
+```bash
+aws ec2 create-vpc-endpoint \
+  --vpc-id <Client VPC ID> \
+  --service-name <VPC Endpoint Service Name> \
+  --subnet-ids <Client Subnets> \
+  --vpc-endpoint-type Interface
+```
+
+Make sure to save the `VpcEndpointId` and the the first DNS Entry `DNSName`.
+
+### Accept the VPC Endpoint in your VPC Endpoint Service
+```bash
+aws ec2 accept-vpc-endpoint-connections \
+  --service-id <VPC Endpoint Service ID> \
+  --vpc-endpoint-ids <VPC Endpoint ID>
+```
+
+### Add Route 53 Private Hosted Zone
+For your client machine to be able to resolve the custom wilcard DNS configured for Zilla, you need to add a Route 53 Private Hosted Zone, with a `CNAME` record that resolved the custom domain to your previously created VPC Endpoint DNS name.
+
+```bash
+aws route53 create-hosted-zone \
+  --name example.aklivity.io \
```

Comment:
Use example with `us-east-1` prefix, as including the region is best practice so we want to lead them there.